include/ppx/scene/scene_pipeline_args.h

```diff
@@ -0,0 +1,219 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ppx_scene_pipeline_args_h
+#define ppx_scene_pipeline_args_h
+
+#include "ppx/scene/scene_config.h"
+#include "ppx/scene/scene_material.h"
+#include "ppx/grfx/grfx_descriptor.h"
+#include "ppx/camera.h"
+
+#include <unordered_map>
+
+namespace ppx {
+namespace scene {
+
+class MaterialPipelineArgs;
+
+// -------------------------------------------------------------------------------------------------
+
+// size = 8
```

Comment:
You might want to use `static_assert(sizeof(X) == N)` and maybe `OFFSET_OF` with static_assert if the layout is important.

Reply:
There is static assert in scene_pipeline_args.cpp for the constants, perhaps that is sufficient?